### PR TITLE
Add sed to fix issue#27 for longer password length

### LINKED ISSUE #27: Add Support for longer passwords by default (64 characters instead of 16) without requiring a device manager
**Security Improvement Feature.**

### PR DESCRIPTION
**Pull request by AkechiShiro**
If there is any misplacement of the sed lines, be sure to tell me where I should move them in the logic of the Patch.sh
Fixes #27

### CODE REVIEW

**Comment by SkewedZeppelin** [comment]
While that file exists in all branches, MAX_PASSWORD_LENGTH is only defined in 18.1 and 19.1

And the Settings change is needed for <18.1 too: https://github.com/GrapheneOS/platform_packages_apps_Settings/commit/0f1fc64287ea8f318b5d921bbfa96df6220a220c

**Comment by AkechiShiro** [comment]
Hey @SkewedZeppelin there were already patches for LOS <18 but up to 48 chars max, I've increased all those to 64 chars.

**Comment by AkechiShiro** [comment]
Hey any reason why this PR was closed ? Should I have signed my commits ?
Also I've seen you added more sed for a return statement I believe, would you mind explaining that please ?

File: Scripts/LineageOS-15.1/Patch.sh [hunk 1]
```diff
@@ -120,6 +120,7 @@ sed -i 's/DEFAULT_MAX_FILES = 1000;/DEFAULT_MAX_FILES = 0;/' services/core/java/
 sed -i 's/DEFAULT_MAX_FILES_LOWRAM = 300;/DEFAULT_MAX_FILES_LOWRAM = 0;/' services/core/java/com/android/server/DropBoxManagerService.java;
 sed -i 's/(notif.needNotify)/(true)/' location/java/com/android/internal/location/GpsNetInitiatedHandler.java; #Notify the user if their location is requested via SUPL
 sed -i 's/DEFAULT_STRONG_AUTH_TIMEOUT_MS = 72 \* 60 \* 60 \* 1000;/DEFAULT_STRONG_AUTH_TIMEOUT_MS = 12 * 60 * 60 * 1000;/' core/java/android/app/admin/DevicePolicyManager.java; #Decrease the strong auth prompt timeout to occur more often
+sed -i 's/mPasswordMaxLength = 16;/mPasswordMaxLength = 64;/' src/com/android/settings/password/ChooseLockPassword.java #MaxPassword Length increased (GrapheneOS)
```
Comment (SkewedZeppelin) [inline Scripts/LineageOS-15.1/Patch.sh:123]: this is the Settings changed placed in the frameworks/base section and does not match the others

File: Scripts/LineageOS-18.1/Patch.sh [hunk 2]
```diff
@@ -173,6 +173,7 @@ sed -i 's/(notif.needNotify)/(true)/' location/java/com/android/internal/locatio
 sed -i 's/entry == null/entry == null || true/' core/java/android/os/RecoverySystem.java; #Skip strict update compatibiltity checks XXX: TEMPORARY FIX
 sed -i 's/!Build.isBuildConsistent()/false/' services/core/java/com/android/server/wm/ActivityTaskManagerService.java; #Disable partition fingerprint mismatch warnings XXX: TEMPORARY FIX
 sed -i 's/DEFAULT_STRONG_AUTH_TIMEOUT_MS = 72 \* 60 \* 60 \* 1000;/DEFAULT_STRONG_AUTH_TIMEOUT_MS = 12 * 60 * 60 * 1000;/' core/java/android/app/admin/DevicePolicyManager.java; #Decrease the strong auth prompt timeout to occur more often
+sed -i 's/MAX_PASSWORD_LENGTH = 16;/MAX_PASSWORD_LENGTH = 64;/' core/java/android/app/admin/DevicePolicyManager.java; # Increase max password length (GrapheneOS)
```
Comment (SkewedZeppelin) [inline Scripts/LineageOS-18.1/Patch.sh:176]: this one and the 19.1 one are correct

File: Scripts/LineageOS-14.1/Patch.sh [hunk 3]
```diff
@@ -220,7 +220,7 @@ if [ "$DOS_SENSORS_PERM" = true ]; then
 applyPatch "$DOS_PATCHES/android_packages_apps_Settings/0002-Sensors-P1.patch"; #Permission for sensors access (MSe1969)
 applyPatch "$DOS_PATCHES/android_packages_apps_Settings/0002-Sensors-P2.patch";
 fi;
-sed -i 's/private int mPasswordMaxLength = 16;/private int mPasswordMaxLength = 48;/' src/com/android/settings/ChooseLockPassword.java; #Increase max password length (GrapheneOS)
+sed -i 's/private int mPasswordMaxLength = 16;/private int mPasswordMaxLength = 64;/' src/com/android/settings/ChooseLockPassword.java; #Increase max password length (GrapheneOS)
```
Comment (SkewedZeppelin) [inline Scripts/LineageOS-14.1/Patch.sh:223]: this and the 16.0 and 17.1 are correct

**Review by SkewedZeppelin — COMMENTED** [review]
OK so:

14/15/16/17 needed this for frameworks/base: https://github.com/GrapheneOS/platform_frameworks_base/commit/0708fbd84b43e43b6879d17090c96a30e4227c4e
and this for packages/apps/Settings: https://github.com/GrapheneOS/platform_packages_apps_Settings/commit/0f1fc64287ea8f318b5d921bbfa96df6220a220c

18 and 19 only needed this for frameworks/base: https://github.com/GrapheneOS/platform_frameworks_base/commit/63da3a29aab246b89653b7fc282bff9d36fbb77e

DivestOS had already included the Settings change for 14/15/16/17, but at 48 instead of 64 characters

And yes commits need to be signed using -s as appropriate: https://divestos.org/index.php?page=about#contributing

**Comment by SkewedZeppelin** [comment]
I appreciate your contribution and I should reviewed it and let you fix it, but I was impatient. Apologies.
I've added you to the website credits regardless and will try to better review if you decide to contribute again.

**Comment by AkechiShiro** [comment]
It's okay, I was just confused as I might have been doing the sed somewhere I wasn't supposed too or else, I understand you could be eager for such an easy fix, it's alright I'll give a try to contribute once more, it's okay if you think that single contribution doesn't warrant me any credit, I didn't start working on helping for fame, I'm looking to run DivestOS as my daily driver on Android in a few months, hence it's in my best interest to help it best as I can.

**Comment by AkechiShiro** [comment]
I'd like to ask is there an IRC chat or any place where we can contact/exchange with tthe DivestOS devs/community about issues or questions, we might have ?

**Comment by SkewedZeppelin** [comment]
https://divestos.org/index.php?page=about#community

**Comment by SkewedZeppelin** [comment]
> be inside the documentation for everyone to see

common/important information should be added to the website so people don't have
to hunt around on GitHub/GitLab.

**Pull request closed without merging**